pkg/controllers/scalablenodegroup/v1alpha1/suite_test.go

```diff
@@ -53,24 +54,76 @@ var _ = AfterSuite(func() {
 
 var _ = Describe("Examples", func() {
 	var ns *environment.Namespace
+	var defaultReplicaCount = ptr.Int32(3)
+
 	var sng *v1alpha1.ScalableNodeGroup
 
 	BeforeEach(func() {
 		var err error
 		ns, err = env.NewNamespace()
 		Expect(err).NotTo(HaveOccurred())
 		sng = &v1alpha1.ScalableNodeGroup{}
+		fakeCloudProvider.NodeGroupStable = true
+		fakeCloudProvider.NodeGroupMessage = ""
+		fakeCloudProvider.WantErr = nil
+		sng.Spec.Replicas = defaultReplicaCount
+		fakeCloudProvider.NodeReplicas[sng.Spec.ID] = ptr.Int32(0)
```

Comment:
if I understand correctly, sng.Spec.ID is nil here, so this line is just muddying the water.

Reply:
`sng.Spec.ID` is an empty string.
This is required here, since, in my tests also `sng.Spec.ID` is always an empty string and NodeReplicas[""] is a Nil value if I don't set it which causes `nodegroup.Replicas` to be also Nil.